src/main/java/com/unascribed/fabrication/FabConf.java

```diff
@@ -0,0 +1,715 @@
+		return config.get(k) == ConfigValue.BANNED;
+	}
+
+	@Environment(EnvType.CLIENT)
+	private static boolean clientCheckBanned(String configKey) {
+		return FabricationModClient.isBannedByServer(configKey);
+	}
+
+	public static boolean isFailed(String configKey) {
+		return failures.contains(remap(configKey));
+	}
+
+	public static ConfigValue getValue(String configKey) {
+		if (isBanned(configKey)) return ConfigValue.BANNED;
+		if (isFailed(configKey)) return ConfigValue.FALSE;
+		ConfigValue worldVal = worldConfig.get(configKey);
+		if (worldVal != null) return worldVal;
+		return config.getOrDefault(remap(configKey), ConfigValue.UNSET);
+	}
+
+	public static boolean doesWorldContainValue(String configKey){
+		return worldConfig.get(configKey) != null || worldDefaults.get(configKey) != null;
+	}
+
+	public static ResolvedConfigValue getResolvedValue(String configKey) {
+		if (isBanned(configKey)) return ResolvedConfigValue.BANNED;
+		if (isFailed(configKey)) return ResolvedConfigValue.FALSE;
+		ConfigValue worldVal = worldConfig.get(configKey);
+		if (worldVal != null) {
+			Boolean worldResult = worldDefaults.get(configKey);
+			return worldVal.resolveSemantically(worldResult != null ? worldResult : defaults != null && defaults.getOrDefault(configKey, false));
+		}
+		return config.getOrDefault(remap(configKey), ConfigValue.UNSET).resolveSemantically(defaults != null && defaults.getOrDefault(configKey, false));
+	}
+
+	public static boolean isStandardValue(String s) {
+		return !NON_STANDARD.contains(s);
+	}
+
+	public static boolean isRuntimeConfigurable(String s) {
+		return RUNTIME_CONFIGURABLE.contains(s);
+	}
+
+	public static String getRawValue(String configKey) {
+		configKey = remap(configKey);
+		return rawConfig.get(configKey).orElse(configKey.equals("general.profile") ? "light" : "");
+	}
+
+	public static Profile getWorldProfile(){
+		return worldProfile;
+	}
+
+	public static boolean isValid(String configKey) {
+		return validKeys.contains(remap(configKey));
+	}
+
+	public static boolean getDefault(String configKey) {
+		return defaults != null && defaults.get(remap(configKey));
+	}
+
+	public static boolean defaultContains(String configKey) {
+		return defaults != null && defaults.containsKey(remap(configKey));
+	}
+
+	public static ImmutableSet<String> getAllKeys() {
+		return validKeys;
+	}
+
+	public static ImmutableSet<String> getAllSections() {
+		return validSections;
+	}
+
+	public static Set<String> getAllFailures() {
+		return failuresReadOnly;
+	}
+
+	public static Collection<String> getAllBanned() {
+		return Collections2.transform(Collections2.filter(config.entrySet(), en ->{
+			ConfigValue worldVal = worldConfig.get(en.getKey());
+			return en.getValue() == ConfigValue.BANNED || worldVal == ConfigValue.BANNED;
+		}), Map.Entry::getKey);
+	}
+
+	public static void addFailure(String configKey) {
+		failures.add(remap(configKey));
+	}
+
+	public static void worldSet(String configKey, String newValue) {
```

Comment:
This should probably be factored out into a private helper method that accepts a Path instead of being a copy paste.